src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -445,6 +446,7 @@ Namespace Microsoft.CodeAnalysis.Editor.Implementation.Diagnostics.UnitTests
                 Assert.Equal(0, diagnostics.Count())
 
                 diagnostics = exceptionDiagnosticsSource.TestOnly_GetReportedDiagnostics(analyzer)
+                ' This fails in V2 because exceptionDiagnosticsSource is not updated. Why?
```

Comment:
The v1 engine ensures that exception diagnostic source is hooked up prior to invoking any Execute**Actions APIs on the AnalyzerExecutor. Currently, CompilationWithAnalyzers just adds all analyzer exception diagnostics to its private field _exceptionDiagnostics. However, for IDE we want to do things such as report non-fatal Watson, etc. So you probably need to add/update public APIs to CompilationWithAnalyzers to accept onAnalyzerException delegate when executing analyzers, which should be hooked up to invoke into the AbstractHostDiagnosticsUpdateSource in the v2 engine and also report non-fatal watsons.